circuits/cpp/barretenberg/cpp/src/barretenberg/bb/main.cpp

```diff
@@ -56,62 +80,153 @@ void prove(const std::string& bytecodePath,
     auto constraint_system = get_constraint_system(bytecodePath);
     auto witness = get_witness(witnessPath);
     auto proof = acir_composer->create_proof(srs::get_crs_factory(), constraint_system, witness, recursive);
+
+    std::cout << proof << std::endl;
     write_file(outputPath, proof);
+
     info("proof written to: ", outputPath);
 }
-
+/**
+ * @brief Computes the number of Barretenberg specific gates needed to create a proof for the specific ACIR circuit
+ *
+ * Communication:
+ * - stdout: The number of gates is written to stdout
+ *
+ * @param bytecodePath Path to the file containing the serialized circuit
+ */
 void gateCount(const std::string& bytecodePath)
 {
     auto acir_composer = new acir_proofs::AcirComposer(MAX_CIRCUIT_SIZE, verbose);
     auto constraint_system = get_constraint_system(bytecodePath);
     acir_composer->create_circuit(constraint_system);
-    info("gates: ", acir_composer->get_total_circuit_size());
+    auto gate_count = acir_composer->get_total_circuit_size();
+    std::cout << gate_count << std::endl;
 }
-
-bool verify(const std::string& proof_path, bool recursive, const std::string& vk_path)
+/**
+ * @brief Verifies a proof for an ACIR circuit
+ *
+ * Note: The fact that the proof was computed originally by parsing an ACIR circuit is not of importance
+ * because this method uses the verification key to verify the proof.
+ *
+ * Communication:
+ * - stdout: A boolean value is printed to stdout indicating whether the proof is valid
+ *
+ * @param proof_path Path to the file containing the serialized proof
+ * @param recursive Whether to use recursive proof generation of non-recursive
+ * @param vk_path Path to the file containing the serialized verification key
+ * @return true If the proof is valid
+ * @return false If the proof is invalid
```

Comment:
```suggestion
```

This function returns `void`

Reply:
Instead of deleting the comment, I'll make it return a bool and then 0 for success and 1 for failure